internal/web_package/injector.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * @license
+ * Copyright 2018 The Bazel Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const parse5 = require('parse5');
+const treeAdapter = require('parse5/lib/tree-adapters/default');
+const fs = require('fs');
+const path = require('path');
+
+function findElementByName(d, name) {
+  if (treeAdapter.isTextNode(d)) return undefined;
+  if (d.tagName && d.tagName.toLowerCase() === name) {
+    return d;
+  }
+  if (!treeAdapter.getChildNodes(d)) {
+    return undefined;
+  }
+  for (let i = 0; i < treeAdapter.getChildNodes(d).length; i++) {
+    const f = treeAdapter.getChildNodes(d)[i];
+    const result = findElementByName(f, name);
+    if (result) return result;
+  }
+  return undefined;
+}
+
+function main(args) {
+  const params = fs.readFileSync(args[0], {encoding: 'utf-8'}).split('\n').filter(l => !!l);
+  const output_file = params.shift();
+  const input_file = params.shift();
+
+  const document = parse5.parse(fs.readFileSync(input_file, {encoding: 'utf-8'}), {treeAdapter});
+  const body = findElementByName(document, 'body');
+  if (!body) {
+    console.error('No <body> tag found in HTML document');
+    return 1;
+  }
+  const head = findElementByName(document, 'head');
+  if (!head) {
+    console.error('No <head> tag found in HTML document');
+    return 1;
+  }
+
+  for (const s of params.filter(s => /\.js$/.test(s))) {
+    const script = treeAdapter.createElement('script', undefined, [
+      {name: 'type', value: 'text/javascript'},
+      {name: 'src', value: `/${path.basename(s)}`},
+    ]);
+    treeAdapter.appendChild(body, script);
+  }
+
+  for (const s of params.filter(s => /\.css$/.test(s))) {
+    const stylesheet = treeAdapter.createElement('link', undefined, [
```

Comment:
This won't add newlines after script/style tags.

Reply:
yes, agreed, I don't think that's required thugh

Reply:
Yep, agreed